modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -56,481 +85,467 @@ GEDI_AGB_prep <- function(site_info, time_points, outdir = file.path(getwd(), "G
   AGB_Output <- matrix(NA, length(site_info$site_id), 2*length(time_points)+1) %>% 
     `colnames<-`(c("site_id", paste0(time_points, "_AGB"), paste0(time_points, "_SD"))) %>% as.data.frame()#we need: site_id, AGB, std, target time point.
   AGB_Output$site_id <- site_info$site_id
-  # loop over each time point
+  # loop over each time point.
   for (i in seq_along(time_points)) {
     # create start and end dates.
     start_date <- seq(time_points[i], length.out = 2, by = paste0("-", search_window))[2]
     end_date <- seq(time_points[i], length.out = 2, by = search_window)[2]
-    # extract GEDI AGB.
-    AGB <- GEDI_AGB_extract(site_info, start_date, end_date, outdir, nfile.min = 0, nrow.min = 1, buffer = as.numeric(buffer))
-    for (j in seq_along(AGB)) {
+    # create the download folder for downloaded GEDI tiles.
+    download.path <- file.path(outdir, "download")
+    if (!dir.exists(download.path)) {
+      dir.create(download.path)
+    } else {
+      # delete previous downloaded files.
+      unlink(download.path, recursive = T)
+      dir.create(download.path)
+    }
+    # download GEDI tiles.
+    files <- NASA_DAAC_download(ul_lat = bbox[4], 
+                                ul_lon = bbox[1], 
+                                lr_lat = bbox[3], 
+                                lr_lon = bbox[2], 
+                                ncore = cores, 
+                                from = start_date, 
+                                to = end_date, 
+                                outdir = download.path, 
+                                doi = "10.3334/ORNLDAAC/2056", 
+                                just_path = F, 
+                                credential.folder = outdir)
```

Comment:
Would it be possible to have GEDI_AGB_prep() accept `credential.file` as an argument as well (possibly defaulting to `"~/.nasadaacapirc"`?) and pass it in here? I wouldn't want to leave credential files sitting  around the outfolder, so being able to pass an in-place path seems preferable to needing to remember to copy first and delete after.

Reply:
Fixed.